pow/src/main/java/tech/pegasys/teku/pow/Web3jEth1Provider.java

```diff
@@ -34,19 +38,29 @@
 import org.web3j.protocol.core.methods.response.EthSyncing;
 import tech.pegasys.teku.infrastructure.async.AsyncRunner;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.time.TimeProvider;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.pow.exception.RejectedRequestException;
 import tech.pegasys.teku.util.config.Constants;
 
-public class Web3jEth1Provider implements Eth1Provider {
+public class Web3jEth1Provider extends AbstractMonitorableEth1Provider {
   private static final Logger LOG = LogManager.getLogger();
 
+  private final AtomicBoolean validating = new AtomicBoolean(false);
+
+  private final String id;
   private final Web3j web3j;
   private final AsyncRunner asyncRunner;
 
-  public Web3jEth1Provider(final Web3j web3j, final AsyncRunner asyncRunner) {
+  public Web3jEth1Provider(
+      final int index,
+      final Web3j web3j,
+      final AsyncRunner asyncRunner,
+      TimeProvider timeProvider) {
+    super(timeProvider);
     this.web3j = web3j;
     this.asyncRunner = asyncRunner;
+    this.id = String.valueOf(index + 1);
```

Comment:
I'm not sure this ID will mean much to users - it would probably make more sense to use the URL of the endpoint (or at least the host and port part of it if that's too long).

Reply:
the ID here encodes the priority of the endpoint (the order in the endpoint list), so it allows the user to know that when a lower ID comes alive again, the system will start to use it (failback behaviour).
I agree to add the endpoint host (the hostname only to avoid unveiling api keys (ie infura)) but i would leave the index so as a user you know recall the priority of that endpoint. Do you agree?

Reply:
ah yeah that's an interesting point.  Good idea.